### ISSUE TITLE
urls are ignored when rendering a QuadMesh to SVG

### ISSUE THREAD

**Issue by pingyeh** [matplotlib/matplotlib#12392]
### Bug report

**Bug summary**

I've called set_urls() to set URLs to a QuadMesh, but when I save it to SVG the URLs are missing.

I traced it to the `RendererBase.draw_quad_mesh()` method which passes `[None]` as the urls argument to `self.draw_path_collection()`.

**Code for reproduction**

```python
import numpy as np
import matplotlib.pyplot as plt

nan = np.nan
data = np.array([[nan, 4, nan], [2, 7, 5], [nan, 8, nan]])
links = np.array([[nan, 'http://abc.com/', nan],
                  ['http://cnn.com/', 'http://nbc.com/', 'http://cbs.com/'],
                  [nan, 'http://nytimes.com',]])

fig = plt.figure()
ax = plt.gca()
mesh = ax.pcolormesh(data, urls=links)
fig.savefig('/tmp/test_links.svg')
```

**Actual outcome**

```
$ fgrep http /tmp/test_link.svg 
  "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd">
<!-- Created with matplotlib (http://matplotlib.org/) -->
<svg height="288pt" version="1.1" viewBox="0 0 432 288" width="432pt" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
```

**Expected outcome**

The URLs are inserted into the quadmesh <g> element like this:
<g id="QuadMesh_1">
  <a xlink:href="url"> <path> /* the path */ </path> </a>
  <a xlink:href="url"> <path> /* the path */ </path> </a>
</g>

**Matplotlib version**

  * Operating system: Linux 
  * Matplotlib version: 2.2.2
  * Matplotlib backend (`print(matplotlib.get_backend())`): module://ipykernel.pylab.backend_inline, but SVG is what I care about
  * Python version: 3.5.3
  * Jupyter version (if applicable): 4.4.0
  * Other libraries: 

pip

**Comment by tacaswell** [comment]
Could you open a pull request to fix that?

It looks like this will require adding an extra argument to `draw_quad_mesh` and propagating that change out to all of the backends?

**Comment by pingyeh** [comment]
Yes, that's probably the smallest change to make it work. I can make a pull request.

**Comment by shouldsee** [comment]
is this fixed on any branch?

**Comment by pingyeh** [comment]
I checked `draw_quad_mesh` briefly and did not find a good way to do it last Oct. Then I got into other work. I don't yet have a working branch for this.

I can make another attempt this month. I may need some guidance though, because I'm not familiar with the C++ code base.

**Comment by tacaswell** [comment]
I don't think this will require going down to the c++ level, at least not for the SVG backend.

**Comment by pingyeh** [comment]
Ah, that should make it easier. I'll try again and report back here.
